### DISCUSSION TITLE
work on module 5.18.0

### DISCUSSION THREAD

**Discussion by SevaSeva17** [morrownr/8814au#83] [Q&A] [answered]
Will the driver work on module 5.18.0?

If not what should I do?

**Comment by morrownr** [comment, marked as answer]
Hi @SevaSeva17 

> Will the driver work on module 5.18.0?

Yes. It is updated all the way to kernel 6.0. It is running here on kernel 6.0 rc3.

**Reply by SevaSeva17** [reply]
Thank u a lot. I've installed the driver but I still can't see avaliable wi-fi networks. Kali Linux shows my network adapter as a connected device. May be u can give me any advice?

**Comment by morrownr** [comment]
How about posting the results of the following commands:

$ lsusb
$ iw dev

FYI: Right now I don't have Kali installed on anything. The last release seemed a little unstable to me as it blew up on two different test systems. It is not like I have not installed Linux a few thousand times. I'm waiting for the next release as I don't have time for things that aren't working well.

**Reply by SevaSeva17** [reply]
1)
lsusb  
Bus 001 Device 003: ID 2357:0106 TP-Link Archer T9UH v1 [Realtek RTL8814AU]
Bus 001 Device 002: ID 80ee:0021 VirtualBox USB Tablet
Bus 001 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
2)
iw dev  
phy#0
	Interface wlan0
		ifindex 3
		wdev 0x1
		addr be:f4:07:74:ab:aa
		type managed
		txpower 20.00 dBm

No one answered me before so u'r already great

**Comment by morrownr** [comment]
This the correct driver and you have a wifi interface. Hmmm...

Let's try the results of some more commands:

$ dkms status
$ iw reg get

Tell me what Linux distro and version you are using.

Tell me what channels you want to connect to.

Regards

**Reply by SevaSeva17** [reply]
I'm using Kali Linux 5.18.0.

I just want to connect to my wi-fi network and then probably try to scan it, for instance.

dkms status
rtl8814au/5.8.5.1, 5.18.0-kali5-amd64, x86_64: installed

iw reg get
global
country 00: DFS-UNSET
	(755 - 928 @ 2), (N/A, 20), (N/A), PASSIVE-SCAN
	(2402 - 2472 @ 40), (N/A, 20), (N/A)
	(2457 - 2482 @ 20), (N/A, 20), (N/A), AUTO-BW, PASSIVE-SCAN
	(2474 - 2494 @ 20), (N/A, 20), (N/A), NO-OFDM, PASSIVE-SCAN
	(5170 - 5250 @ 80), (N/A, 20), (N/A), AUTO-BW, PASSIVE-SCAN
	(5250 - 5330 @ 80), (N/A, 20), (0 ms), DFS, AUTO-BW, PASSIVE-SCAN
	(5490 - 5730 @ 160), (N/A, 20), (0 ms), DFS, PASSIVE-SCAN
	(5735 - 5835 @ 80), (N/A, 20), (N/A), PASSIVE-SCAN
	(57240 - 63720 @ 2160), (N/A, 0), (N/A)

phy#0
country 99: DFS-UNSET
	(2402 - 2472 @ 40), (N/A, 20), (N/A)
	(2457 - 2482 @ 40), (N/A, 20), (N/A), PASSIVE-SCAN
	(5140 - 5860 @ 40), (N/A, 30), (N/A), PASSIVE-SCAN

**Comment by morrownr** [comment]
> country 00: DFS-UNSET

Your country code is not set. What country are you in?

**Reply by SevaSeva17** [reply]
Russia

**Comment by morrownr** [comment]
$ sudo iw reg set RU

...and then give it a try.

**Reply by SevaSeva17** [reply]
iw reg get
global
country RU: DFS-UNSET
	(2400 - 2483 @ 40), (N/A, 20), (N/A)
	(5150 - 5350 @ 160), (N/A, 20), (N/A), NO-OUTDOOR
	(5650 - 5850 @ 160), (N/A, 20), (N/A), NO-OUTDOOR
	(57000 - 66000 @ 2160), (N/A, 40), (N/A), NO-OUTDOOR

phy#0
country 99: DFS-UNSET
	(2402 - 2472 @ 40), (N/A, 20), (N/A)
	(2457 - 2482 @ 40), (N/A, 20), (N/A), PASSIVE-SCAN
	(5140 - 5860 @ 40), (N/A, 30), (N/A), PASSIVE-SCAN

Now it shows I'm russian but I still don't see any wi-fi networks

**Comment by morrownr** [comment]
Let's try:

$ nmcli dev wifi

...and see what the result is.

**Reply by SevaSeva17** [reply]
I'm really sorry I couldn't anwer earilier 

Here is the result

nmcli dev wifi
IN-USE  BSSID  SSID  MODE  CHAN  RATE  SIGNAL  BARS  SECURITY

**Comment by morrownr** [comment]
Hmmm... it appears that no scanning is taking place.

Post the results of:

$ rfkill list all

**Reply by SevaSeva17** [reply]
rfkill list all
0: phy0: Wireless LAN
	Soft blocked: no
	Hard blocked: no

**Comment by morrownr** [comment]
Okay, so Airplane mode is not on.

I'm running out of ideas. The driver is installed and a wifi interface is being created. That means the problem areas should be okay. We are down to looking at things like rebooting the system or trying a different USB port. Do you have the ability to try on another computer or install another os on the system you are using as a dual boot?

Let me point out the Main Menu for this site as it has a lot of information:

https://github.com/morrownr/USB-WiFi

Regards

**Reply by SevaSeva17** [reply]
I can try to install Kali Linux to VMware Workstation instead of VirtualBox. Thank a lot for ur attemptions to help anyway.

**Comment by morrownr** [comment]
Oh my. Did you mention Virtualbox? I guess I missed it. That makes a lot of difference.

In the FAQ, there is a link to an article that helps you set up things for Virtualbox.

**Reply by SevaSeva17** [reply]
Actually I missed it I'm really sorry then

Edit: I've done it before I asked u to help. When I sent the result of 'lsusb' u could see my network adapter

**Comment by morrownr** [comment]
> When I sent the result of 'lsusb' u could see my network adapter

I simply don't have time to go over all of the information that is available so I concentrate on what I am looking for. I saw the line about your adapter but overlooked the line about Virtualbox.

Good luck.
